aea/cli_gui/app.py

```diff
@@ -0,0 +1,250 @@
+
+    return items_list
+
+
+def get_registered_skills():
+    """Get the registered skills."""
+    work_dir = os.path.join(os.getcwd())
+    skills_dir = os.path.join(work_dir, "packages/skills")
+
+    # Get a list of all the directories paths that ends with .txt from in specified directory
+    file_list = glob.glob(os.path.join(skills_dir, '*'))
+
+    items_list = []
+
+    for path in file_list:
+        if is_skill_dir(path):
+            head, tail = os.path.split(path)
+            items_list.append({"id": tail, "description": "placeholder description"})
+
+    return items_list
+
+
+def call_aea(param_list, dir):
+    """Call the cli commands."""
+    old_cwd = os.getcwd()
+    os.chdir(dir)
+
+    ret = subprocess.call(param_list)
+    os.chdir(old_cwd)
+    return ret
+
+
+def create_agent(agent_id):
+    """Create the agent from GUI."""
+    if call_aea(["aea", "create", agent_id], args.agent_dir) == 0:
+        return agent_id, 201  # 201 (Created)
+    else:
+        return {"detail": "Failed to create Agent {} - a folder of this name may exist already".format(
+            agent_id)}, 400  # 400 Bad request
+
+
+def delete_agent(agent_id):
+    """Delete the agent from GUI."""
+    if call_aea(["aea", "delete", agent_id], args.agent_dir) == 0:
+        return 'Agent {} deleted'.format(agent_id), 200  # 200 (OK)
+    else:
+        return {"detail": "Failed to delete Agent {} - it ay not exist".format(agent_id)}, 400  # 400 Bad request
+
+
+def fetch_item(agent_id, item_type, item_id):
+    """Fetch the items from the packages folder."""
+    dir = os.path.join(args.agent_dir, agent_id)
+    if call_aea(["aea", "add", item_type, item_id], dir) == 0:
+        return agent_id, 201  # 200 (OK)
+    else:
+        return {"detail": "Failed to add protocol {} to agent {}".format(item_id, agent_id)}, 400  # 400 Bad request
+
+
+def remove_local_item(agent_id, item_type, item_id):
+    """Remove a local item."""
+    dir = os.path.join(args.agent_dir, agent_id)
+    if call_aea(["aea", "remove", item_type, item_id], dir) == 0:
+        return agent_id, 201  # 200 (OK)
+    else:
+        return {"detail": "Failed to remove protocol {} from agent {}".format(item_id,
+                                                                              agent_id)}, 400  # 400 Bad request
+
+
+def get_local_items(agent_id, item_type):
+    """Get the local items."""
+    dir = os.path.join(os.path.join(args.agent_dir, agent_id), item_type + "s")
+
+    # Get a list of all the directories paths that ends with .txt from in specified directory
+    file_list = glob.glob(os.path.join(dir, '*'))
+
+    items_list = []
+
+    for path in file_list:
+        if is_item_dir(path, item_type):
+            head, tail = os.path.split(path)
+            items_list.append({"id": tail, "description": "placeholder description"})
+
+    return items_list
+
+
+app = connexion.FlaskApp(__name__, specification_dir='./')
```

Comment:
`specification_dir='./'` can make the script to fail in all the circumstances where the `app.py` script is not executed from `aea/cli_gui`. 
So I'd suggest to avoid the relative path and use the absolute path of the file `app.py`.

This can be done in the following way:
```
import os
import inspect
CUR_DIR = os.path.dirname(inspect.getfile(inspect.currentframe()))
...
 app = connexion.FlaskApp(__name__, specification_dir=CUR_DIR)
```